### ISSUE TITLE
DataList index not refresh

### ISSUE THREAD

**Issue by LucasAlumni** [primefaces/primeng#4434]
### There is no guarantee in receiving a response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeNG PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
Index row not refresh after array.splice (immutable="false") and array.slice (immutable="true")

`delete(item: Object, name: string, index: any) {
    this.data.splice(index, 1)
  }`

`delete(item: Object, name: string, index: any) {
    this.deleteItem.emit({data: {...item}, name, index})
    this.data = [...this.data.slice(0, index), ...this.data.slice(index + 1)]
  }`
 
**Expected behavior**

**Please tell us about your environment:**
Windows 10, angular-cli (ng-serve)

* **Angular version:** 4.4.4

* **PrimeNG version:** 4.3.0

* **Browser:** Chrome

 
* **Language:** TypeScript 2.3

**Comment by cagataycivici** [comment]
Should be fixed with PrimeNG 5.0.0-RC

**Comment by LucasAlumni** [comment]
Hi,

still not fixed in version 5.0.2
